generated/nidaqmx/_library_interpreter.py

```diff
@@ -6457,18 +6484,21 @@ def set_wfm_attr_callback(
             t0_array,
             dt_array,
             0 if t0_array is None else t0_array.size,
-            self._get_wfm_attr_callback_ptr(set_wfm_attr_callback),
+            wfm_attr_callback,
             None,
             read_array,
             read_array.size,
             ctypes.byref(samps_per_chan_read),
             None,
         )
 
-        timestamps = [_T0_EPOCH + ht_timedelta(seconds=t0 * _INT64_WFM_SEC_PER_TICK) for t0 in t0_array]
-        sample_intervals = [ht_timedelta(seconds=dt * _INT64_WFM_SEC_PER_TICK) for dt in dt_array]
+        if timestamps is not None and t0_array is not None:
+            timestamps[:] = [_T0_EPOCH + ht_timedelta(seconds=t0 * _INT64_WFM_SEC_PER_TICK) for t0 in t0_array]
 
-        return error_code, samps_per_chan_read.value, timestamps, sample_intervals
+        if sample_intervals is not None and dt_array is not None:
+            sample_intervals[:] = [ht_timedelta(seconds=dt * _INT64_WFM_SEC_PER_TICK) for dt in dt_array]
```

Comment:
On further thought: this is doing unnecessary list allocations and copies. It would be better to allocate the numpy arrays in the caller and convert to hightime in the for-loop that allocates the timing objects.